docker-images/kafka/scripts/kafka_config_generator.sh

```diff
@@ -19,10 +19,23 @@ LISTENER_SECURITY_PROTOCOL_MAP="REPLICATION:SSL"
 SASL_ENABLED_MECHANISMS=""
 
 if [ "$KAFKA_CLIENT_ENABLED" = "TRUE" ]; then
-  LISTENERS="${LISTENERS},CLIENT://0.0.0.0:9092"
-  ADVERTISED_LISTENERS="${ADVERTISED_LISTENERS},CLIENT://$(hostname -f):9092"
 
-  if [ "$KAFKA_CLIENT_AUTHENTICATION" = "scram-sha-512" ]; then
+
+  if [ "$KAFKA_CLIENT_AUTHENTICATION" = "sasl_plaintext" ]; then
```

Comment:
User can configure the authentication on 3 different interfaces:
* For the plaintext / unencrypted (port 9092, this section of the code)
* For the internal TLS interface (see line 76 in this file)
* For the external listener (see line 139 in this file)

You need to add similar section to this to all the sections to let users configure it everywhere.